networking_ccloud/tools/netbox_config_gen.py

```diff
@@ -400,12 +407,35 @@ def main():
     parser.add_argument("-r", "--region", required=True)
     parser.add_argument("-v", "--verbose", action="store_true")
     parser.add_argument("-u", "--switch-user", help="Default switch user", required=True)
-    parser.add_argument("-p", "--switch-password", help="Default switch password", required=True)
+    parser.add_argument("-p", "--switch-password", help="Default switch password")
+    parser.add_argument('-V', '--vault-ref',
+                        help="Instead of a password, use this vault references, formatted like <path>:<field>")
+
+    parser.add_argument('-w', '--wrap-in', help='Keys under which the generated config should be nested under. '
+                                                'Format should be: <key1>/<key2>...')
     parser.add_argument("-s", "--shell", action="store_true")
     parser.add_argument("-o", "--output")
 
     args = parser.parse_args()
 
+    if args.switch_password and args.vault_ref:
+        parser.error("You may only use one of '--vault-ref' or '--switch-passsword'")
+    if not (args.switch_password or args.vault_ref):
+        parser.error("Either'--vault-ref' or '--switch-passsword' must be set")
+
+    if args.vault_ref:
+        args.switch_password = VAULT_REF_REPLACEMENT
+        m = re.match(r'^(\S+)\W*:\W*(\S+)', args.vault_ref)
+        if m:
+            vault_path, vault_field = m.group(1), m.group(2)
+        else:
+            parser.error(f'Invalid vault reference should be <path>:<field>, got {args.vault_ref}')
+
+    if args.wrap_in:
+        m = re.match(r'^(\w+\/)*(\w+)$', args.wrap_in)
```

Comment:
I was a bit confused by the two (), as I thought you wanted to grab something out of the string. How abut adding `?:` if you just want to use them only for grouping? Also, why does the `/` need escaping?

How about `r'^(?:\w+/)*\w+$'`?